rotkehlchen/chain/base/modules/echo/decoder.py

```diff
@@ -0,0 +1,174 @@
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.structures import (
+    DEFAULT_DECODING_OUTPUT,
+    ActionItem,
+    DecoderContext,
+    DecodingOutput,
+)
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
+from rotkehlchen.constants.resolver import evm_address_to_identifier
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChecksumEvmAddress, EvmTokenKind, EvmTransaction
+from rotkehlchen.utils.misc import bytes_to_address
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.chain.evm.structures import EvmTxReceiptLog
+    from rotkehlchen.history.events.structures.evm_event import EvmEvent
+    from rotkehlchen.user_messages import MessagesAggregator
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class EchoDecoder(DecoderInterface):
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator') -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+
+    def _decode_fee_paid(self, context: DecoderContext) -> DecodingOutput:
+        if context.tx_log.topics[0] != FEE_PAID:
+            return DEFAULT_DECODING_OUTPUT
+        if not self.base.is_tracked(bytes_to_address(context.tx_log.topics[1])):
+            return DEFAULT_DECODING_OUTPUT
+
+        for event in context.decoded_events:
+            if (
+                event.sequence_index == context.tx_log.log_index - 1 and
+                event.event_type == HistoryEventType.INFORMATIONAL and
+                event.event_subtype == HistoryEventSubType.APPROVE
+            ):
+                fee_amount = asset_normalized_value(
+                    amount=int.from_bytes(context.tx_log.data[0:32]),
+                    asset=event.asset,
+                )
+                new_event = self.base.make_event_from_transaction(
+                    transaction=context.transaction,
+                    tx_log=context.tx_log,
+                    event_type=HistoryEventType.SPEND,
+                    event_subtype=HistoryEventSubType.FEE,
+                    asset=event.asset,
+                    balance=Balance(amount=fee_amount),
+                    address=FUNDING_CONDUIT,
+                    notes=f'Paid {fee_amount} USDC as part of funding an Echo deal',
+                    counterparty=CPT_ECHO,
+                )
+                context.decoded_events.append(new_event)
+                break
+        return DEFAULT_DECODING_OUTPUT
```

Comment:
leave one line of space

```suggestion
                break

        return DEFAULT_DECODING_OUTPUT
```